private/buf/bufformat/formatter.go

```diff
@@ -2197,6 +2225,36 @@ func (f *formatter) leadingCommentsContainBlankLine(n ast.Node) bool {
 	return newlineCount(info.LeadingWhitespace()) > 1
 }
 
+func (f *formatter) importHasComment(importNode *ast.ImportNode) bool {
+	return f.nodeHasComment(importNode.Keyword) ||
+		f.nodeHasComment(importNode.Name) ||
+		f.nodeHasComment(importNode.Semicolon) ||
+		f.nodeHasComment(importNode.Public) ||
+		f.nodeHasComment(importNode.Weak)
+}
+
+func (f *formatter) nodeHasComment(node ast.Node) bool {
+	if node == nil {
+		return false
+	}
+	nodeinfo := f.fileNode.NodeInfo(node)
+	return nodeinfo.LeadingComments().Len() > 0 ||
+		nodeinfo.TrailingComments().Len() > 0
+}
+
+// importLevel maps import types to a level, so it can be compared and sorted.
+// `import public`=3, `import`=2; `import weak` =1
+func importLevel(node *ast.ImportNode) int {
+	switch {
+	case node.Public != nil:
+		return 3
+	case node.Weak != nil:
+		return 1
+	default:
+		return 2
```

Comment:
The default import level should take precedence, so the precedence is `default, public, weak`.